### ISSUE TITLE
Merging a pull request should trigger a event for the webhooks

### ISSUE THREAD

**Issue by hhenkel** [gogits/gogs#1573]
When a pull request is merged into a repository a commit event should be triggered via the configured webhooks.

**Comment by Unknwon** [comment]
GitHub treats Pull Request differently from a git push, so it has a Pull Request type webhook, I think we should follow this principle.

How do you think?

**Comment by hhenkel** [comment]
I agree with you that it would be the best to have a pull request event for webhooks - that is what I'm hoping in the long run as one could integrate better with other tools like jenkins for example, as it would allow to test pull request before merging. (Which is also possible with github and travis). I guess that is also relevant for #813?

Nonetheless ff such a hook is available it is to be discussed, if there is a need for a commit event via a webhook in the case of a merge. My currently opinion is, that the commit shows up in the commit history and therefore there should be an event be triggered as one might only subscribe to commit events.

**Comment by xPaw** [comment]
GitHub calls both hooks. Pull request hook with action "merged", and then push hook with merge commit.

**Comment by bhchew** [comment]
+1

**Comment by Unknwon** [comment]
`push` event hook has been added.

**Comment by Unknwon** [comment]
Note: https://developer.github.com/v3/activity/events/types/#pullrequestevent

**Comment by lstahlman** [comment]
Any work done on this already?  If not I could work on it in conjunction with a pull request API extension I'm playing with in my test environment.

**Comment by Unknwon** [comment]
@lstahlman no, don't send PR for this.

**Comment by Unknwon** [comment]
Merge this issue into https://github.com/gogits/gogs/issues/2246 , since `push` is already triggered.

**Comment by jayeshkh007** [comment]
Please let me know how can I build Jenkins job once PR merge with master branch. I want to remove PR number s3 bucket once PR merge with master using github publisher plugin.

**Comment by paladugu18** [comment]
@jayeshkh007 I am looking into the same thing, how to kick off a jenkins job once the pull request is merged to master. Essentially i want the code to be deployed to s3 bucket only after the pull request is merged to the master branch. Am curious how you resolved this issue? Any help is appreciated. Thanks.

**Comment by jayeshkhairnar** [comment]
@paladugu18 If you want to build Job once PR merge then you can use "GitHub hook trigger for GITScm polling" plugin, please make sure you webhook configure correctly and integrate with Jenkins.
